### ISSUE TITLE
Experimental data are low

### ISSUE THREAD

**Issue by AHPUymhd** [javyduck/ChatScene#44]
I followed the official tutorial and ran the experiment for scenario 2，
first，python scripts/run_train.py --agent_cfg=adv_scenic.yaml --scenario_cfg=train_scenario_scenic.yaml --mode train_scenario --scenario_id 2
second，python scripts/run_train.py --agent_cfg=adv_scenic.yaml --scenario_cfg=train_agent_scenic.yaml --mode train_agent --scenario_id 2
finally，python scripts/run_eval.py --agent_cfg=adv_scenic.yaml --scenario_cfg=eval_scenic.yaml --mode eval --scenario_id 2 --test_epoch -1
But I have two more critical questions：
1.The results of the evaluation experiment are low, and the collision rate is high.
2.The route generated in the evaluation phase is only 0, so according to the method in the paper, shouldn't the last two routes 8 and 9 be used for evaluation, and why is the result of the evaluation generated being route 0?
We look forward to hearing from you, thank you

**Comment by AHPUymhd** [comment]
@javyduck

**Comment by javyduck** [comment]
Hi, there are some issues for your operation,

1. the first/ second step is correct, the third step is wrong with the parameters "--test_epoch -1", you need to specify the epoch of the ckpt you wanna load from the second step (e.g. "--test_epoch 100").

2. we use route 8 and 9 for evaluation, could you help provide more details about how you get route "0" here. Specifically, the routes for evaluation comes from the setting in "safebench/scenario/config/eval_scenic.yaml", where it specifies [8,9] there.

If you have any further problem, please let me know!

**Comment by AHPUymhd** [comment]
Thank you very much for your answer, but I noticed this phenomenon: I followed the instructions to first train the scenarios and then used these scenarios to train the agent. However, during evaluation, although the agent does not collide with vehicles, it moves a little near the starting point and then stops. What could be the reason for this? Is it because I did not train the agent long enough in the second step? I trained the agent until model.sac.0343.torch; is this training amount insufficient to cause the issue?

**Comment by AHPUymhd** [comment]
@javyduck

**Comment by javyduck** [comment]
In fact, you should do the opposite — the model has likely been overtrained, leading to reward hacking behavior. Instead, consider applying early stopping during training. For example, evaluate using an earlier checkpoint, such as at epoch 50 or epoch 100.

**Comment by Dogaska** [comment]
Hello all, did you manage to resolve the 'stopping behavior' issue @AHPUymhd ? I have the same problem, and I trained an ego agent for 250 epochs with SAC, and it learned to stop. I evaluate its performance for earlier epochs (50, 100,150). It doesn't learn not to collide or brake when needed, and eval performance is low. Do you have any further suggestions to solve this, @javyduck ?

**Comment by javyduck** [comment]
that's kind of weird, which scenario you are traininining on? if @AHPUymhd also has the same issue, i will double check it on my side.

**Comment by Dogaska** [comment]
@javyduck I am training on scenario 8. Also, I run an evaluation using your pretrained ego agent (python scripts/run_eval.py --agent_cfg=adv_scenic.yaml --scenario_cfg=eval_scenic.yaml --mode eval --scenario_id 8 --test_epoch -1) without adding any adversarial agents to the scenario to see how the trained ego will behave. If I am not misunderstanding, it has learned to stop at the intersection only. It is kind of an overfitting, the ego agent memorized to stop at the intersection only to avoid any collision. Here is the record:

https://github.com/user-attachments/assets/dea99072-81ae-4382-a157-c4e306cec4a6

**Comment by javyduck** [comment]
I see — thanks for bringing this to my attention!

There are many potential ways to fix it (e.g., lower down learning rate, adjust the buffer setting, or adjust the reward func), but at this stage, one simple ablation study you could run is to train the agent using only the similar routes as you show in the video (i.e., exclude all other dissimilar routes during training). If the agent then learns to avoid the collision, it would suggest that the issue stems from route imbalance during RL training.

**Comment by javyduck** [comment]
hi @Dogaska have you solved this problem? if not i will check it on my side : )

**Comment by Dogaska** [comment]
Hi @javyduck , I haven't solved it yet, unfortunately.
